### PR TITLE
Patch @bazel/concatjs to run karma tests outside sandbox and fix test runtime errors.

### PR DESCRIPTION
**Pull request by bmd3k**
* Motivation

  Karma tests are failing in chrome in certain environments (including, most recently, some internal Google machines) with errors like:
  
  ```
  [1019/144854.100947:WARNING:gpu_process_host.cc(1228)] The GPU process has crashed 1 time(s)
  [1019/144854.140417:WARNING:gpu_process_host.cc(1228)] The GPU process has crashed 2 time(s)
  [1019/144854.174092:WARNING:gpu_process_host.cc(1228)] The GPU process has crashed 3 time(s)
  [1019/144854.241059:WARNING:gpu_process_host.cc(1228)] The GPU process has crashed 4 time(s)
  [1019/144854.253018:WARNING:gpu_process_host.cc(1228)] The GPU process has crashed 5 time(s)
  [1019/144854.265413:WARNING:gpu_process_host.cc(1228)] The GPU process has crashed 6 time(s)
  [1019/144854.265453:FATAL:gpu_data_manager_impl_private.cc(439)] GPU process isn't usable. Goodbye.
  ```

* Workaround:

  We haven't been able to root cause the issue (is it specific to more recent Chrome versions? Is it OS-specific?) but we do know that we can workaround the issue by launching the karma chrome browser with the '--no-sandbox' option.
  
  This option is widely used by other karma clients. Notably we find it in:
  * Angular's guide on testing with Karma: https://angular.io/guide/testing
  * Google's internal browser launcher definition for chrome-linux (Googlers see https://source.corp.google.com/piper///depot/google3/testing/web/browsers/chrome-linux.json)
  
  So it seems appropriate for us to use this option as well.

* Implementation:

  The most straightforward way to force the '--no-sandbox' option is not very straightforward at all: Patch @bazel/concatjs to use the option. We do this using the [patch-package](https://www.npmjs.com/package/patch-package) library. We update @bazel/concatjs' 'karma.conf.js' file to use '--no-sandbox' even when it is supposed to be launching a Chrome browser with sandbox:

  * https://github.com/bazelbuild/rules_nodejs/blob/5.7.0/packages/concatjs/web_test/karma.conf.js#L387

* Alternatives:

  I examined the Chrome codebase and played around with other options (especially gpu-specific options like '--disable-gpu') but did not find any other Chrome flag that seemed to solve the issue.

  I explored specifying our own karma.conf.js file to define our own browsers or custom launchers but @bazel/concatjs does not allow us to specify these ourselves. It resets these values here:
  * https://github.com/bazelbuild/rules_nodejs/blob/5.7.0/packages/concatjs/web_test/karma.conf.js#L342

  And forces us to use their own custom launchers.

### CODE REVIEW

File: patches/BUILD [hunk 1]
```diff
@@ -0,0 +1 @@
+licenses(["notice"])
```
Comment (JamesHollyer) [inline patches/BUILD:1]: No idea what this does.

**Review by rileyajones — APPROVED** [review]
This generally seems reasonable especially as a short term fix.
As a long term solution have you considered making a PR on the [Concatjs repo](https://github.com/bazelbuild/rules_nodejs)?
[Here is their karma config](https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/concatjs/web_test/karma.conf.js#L382)

**Review by JamesHollyer — COMMENTED** [review]
Why are there so many yarn.lock changes?

**Comment by JamesHollyer** [comment]
> 

yea maybe this function which determines if the --no-sandbox flag is enabled or not is out of date. https://github.com/bazelbuild/rules_nodejs/blob/cd6cecfd6e923bbf782884b68e9842a7bac4d08e/packages/concatjs/web_test/karma.conf.js#L69

**Comment by JamesHollyer** [comment]
> This generally seems reasonable especially as a short term fix. As a long term solution have you considered making a PR on the [Concatjs repo](https://github.com/bazelbuild/rules_nodejs)? [Here is their karma config](https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/concatjs/web_test/karma.conf.js#L382)

Yea maybe [this function](https://github.com/bazelbuild/rules_nodejs/blob/cd6cecfd6e923bbf782884b68e9842a7bac4d08e/packages/concatjs/web_test/karma.conf.js#L69) which determines if the --no-sandbox flag is enabled or not is out of date.

File: patches/BUILD (hunk 1 above)
Comment (bmd3k) [reply patches/BUILD]: Same, no idea what it does. Removed. (But I had to keep the BUILD file)

**Comment by bmd3k** [comment]
> This generally seems reasonable especially as a short term fix. As a long term solution have you considered making a PR on the [Concatjs repo](https://github.com/bazelbuild/rules_nodejs)? [Here is their karma config](https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/concatjs/web_test/karma.conf.js#L382)

I had considered that but, unfortunately, without knowing the root cause there isn't really a clean change to propose. I agree with James that `supportChromeSandboxing()` would be the ideal place to make a change but I don't have a clear idea what that change would be.

**Comment by bmd3k** [comment]
> Why are there so many yarn.lock changes?

Because I added two new yarn dependencies: patch-package and postinstall-postinstall. I used the guide at https://www.npmjs.com/package/patch-package.

**Comment by bmd3k** [comment]
> > This generally seems reasonable especially as a short term fix. As a long term solution have you considered making a PR on the [Concatjs repo](https://github.com/bazelbuild/rules_nodejs)? [Here is their karma config](https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/concatjs/web_test/karma.conf.js#L382)
> 
> I had considered that but, unfortunately, without knowing the root cause there isn't really a clean change to propose. I agree with James that `supportChromeSandboxing()` would be the ideal place to make a change but I don't have a clear idea what that change would be.

It's probably worth noting a couple more things:
* The patch being applied here is to that exact file in the concatjs repo (karma.conf.js).
* patch-package is recommended by rules_nodejs as a means of making these sorts of custom changes if a PR to the main repo isn't possible (https://bazelbuild.github.io/rules_nodejs/changing-rules.html#patching-the-npm-packages)

**Review by JamesHollyer — APPROVED** [review]

**Pull request merged**